app/src/models/commit.ts

```diff
@@ -4,6 +4,22 @@ import { GitAuthor } from './git-author'
 import { GitHubRepository } from './github-repository'
 import { getDotComAPIEndpoint } from '../lib/api'
 
+/** Grouping of information required to create a commit */
+export interface ICommitContext {
+  /**
+   * The summary of the commit message (required)
+   */
+  readonly summary: string
+  /**
+   * Additional details for the commit message (optional)
+   */
+  readonly description: string | null
+  /**
+   * Trailers representing co-author information (optional)
```

Comment:
Technically speaking these trailers could be anything that conforms to the trailer format, not just co-author trailers. Could we say something like

```suggestion
   * An optional array of commit trailers (for example Co-Authored-By trailers) which will be appended to the commit message in accordance with the Git trailer configuration.
```